mdata/ifaces.go

```diff
@@ -12,6 +13,12 @@ type Metrics interface {
 
 type Metric interface {
 	Add(ts uint32, val float64)
-	Get(from, to uint32) (uint32, []chunk.Iter)
-	GetAggregated(consolidator consolidation.Consolidator, aggSpan, from, to uint32) (uint32, []chunk.Iter)
+	Get(from, to uint32) MetricResult
+	GetAggregated(consolidator consolidation.Consolidator, aggSpan, from, to uint32) MetricResult
+}
+
+type MetricResult struct {
```

Comment:
this type is not an interface, so should go into another file, maybe metricresult.go ? actually, since the relation to metrics is already implied due to it being in mdata, I think we can simply call this `Result`